atlasdb-perf/src/main/java/com/palantir/atlasdb/performance/benchmarks/ResultConsumer.java

```diff
@@ -0,0 +1,31 @@
+/**
+ * Copyright 2017 Palantir Technologies
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.atlasdb.performance.benchmarks;
+
+/** This class can be used to "process" the result of a call
+ *  without storing the whole thing in memory.
+ *  'Count' can be used to verify the size of the result
+ *  and 'hash' can be fed into the JMH black hole in the end.
+ */
+public class ResultConsumer {
+    public int count = 0;
+    public int hash = 0;
+
+    public void consume(Object e) {
+        count += 1;
+        hash += e.hashCode();
```

Comment:
Honestly the volatile read is probably the least of our worries and we should just `iter.forEachRemaining(blackhole::consume);` to avoid DCE, but if we really see overhead and want to keep this might want to`hash += System.identityHashCode(loadedCells);` which will at least avoid potential issue if the object has an expensive hashCode.